src/temporal-resolve/passes/TemporalResolvePass.js

```diff
@@ -0,0 +1,151 @@
+		this.scene = scene;
+		this.camera = camera;
+
+		this.renderTarget = new WebGLRenderTarget( 1, 1, {
+			minFilter: LinearFilter,
+			magFilter: LinearFilter,
+			type: FloatType,
+			depthBuffer: false,
+		} );
+
+		this.depthRenderTarget = new WebGLRenderTarget( 1, 1, {
+			minFilter: NearestFilter,
+			magFilter: NearestFilter,
+		} );
+
+		this.velocityPass = new VelocityPass( scene, camera );
+
+		this.fullscreenMaterial = new TemporalResolveMaterial();
+
+		this.fullscreenMaterial.velocityTexture = this.velocityPass.renderTarget.texture;
+		this.fullscreenMaterial.depthTexture = this.depthRenderTarget.texture;
+
+		this.fsQuad = new FullScreenQuad( null );
+		this.fsQuad.material = this.fullscreenMaterial;
+
+		this.setSize( window.innerWidth, window.innerHeight );
+
+	}
+
+	dispose() {
+
+		this.renderTarget.dispose();
+		this.accumulatedSamplesTexture.dispose();
+		this.lastDepthTexture.dispose();
+		this.fullscreenMaterial.dispose();
+
+	}
+
+	setSize( width, height ) {
+
+		this.renderTarget.setSize( width, height );
+		this.depthRenderTarget.setSize( width, height );
+		this.velocityPass.setSize( width, height );
+
+		this.createFramebuffers( width, height );
+
+	}
+
+	createFramebuffers( width, height ) {
+
+		if ( this.accumulatedSamplesTexture )
+			this.accumulatedSamplesTexture.dispose();
+		if ( this.lastDepthTexture ) this.lastDepthTexture.dispose();
+
+		this.accumulatedSamplesTexture = new FramebufferTexture(
+			width,
+			height,
+			RGBAFormat
+		);
+		this.accumulatedSamplesTexture.minFilter = LinearFilter;
+		this.accumulatedSamplesTexture.magFilter = LinearFilter;
+		this.accumulatedSamplesTexture.type = FloatType;
+
+		this.lastDepthTexture = new FramebufferTexture( width, height, RGBAFormat );
+		this.lastDepthTexture.minFilter = NearestFilter;
+		this.lastDepthTexture.magFilter = NearestFilter;
+
+		this.fullscreenMaterial.accumulatedSamplesTexture = this.accumulatedSamplesTexture;
+		this.fullscreenMaterial.lastDepthTexture = this.lastDepthTexture;
+
+		this.fullscreenMaterial.needsUpdate = true;
+
+	}
+
+	render( renderer ) {
+
+		// render depth
+		this.scene.overrideMaterial = meshDepthMaterial;
+		renderer.setRenderTarget( this.depthRenderTarget );
+		renderer.clear();
+
+		const { background } = this.scene;
+		this.scene.background = blackColor;
+
+		renderer.render( this.scene, this.camera );
+
+		this.scene.background = background;
+		this.scene.overrideMaterial = null;
+
+		// render velocity
+		this.velocityPass.render( renderer );
+
+		// update uniforms of this pass
+		this.fullscreenMaterial.tileCount = this.ptRenderer.tiles.x * this.ptRenderer.tiles.y;
+		this.fullscreenMaterial.curInverseProjectionMatrix.copy(
+			this.camera.projectionMatrixInverse
+		);
+		this.fullscreenMaterial.curCameraMatrixWorld.copy(
+			this.camera.matrixWorld
+		);
+		this.fullscreenMaterial.cameraNear = this.camera.near;
+		this.fullscreenMaterial.cameraFar = this.camera.far;
+
+		// now render this fullscreen pass
```

Comment:
I imagine at some point it would be a good plan to switch to use MRT for this, right? Maybe it's worth making an issue for?

Reply:
I think we should drop the depth pass altogether because depth is also saved to the blue channel of the velocity pass. Since the velocity pass uses half float type, the precision shouldn't degrade much when fetching depth (currently it's 32 bit but 16 bit will be okay too).

Reply:
I also noticed that MRT can be slower when rendering 2 half float type textures in one pass vs. rendering one unsigned byte type pass and one half float type pass separately, so I'd have to check the performance between these two options.

Reply:
Gotcha good to know. I didn't know that MRT might be slower if rendering to higher depth textures. I don't think we have to deal with that now. But I did make https://github.com/gkjohnson/three-gpu-pathtracer/issues/253 to track it in the future.